frontend/src/data/repositories/questionRepository.ts

```diff
@@ -0,0 +1,264 @@
+import { IQuestionInput, IQuestionUpdateInput } from '../../domain/repositories/IQuestionRepository';
+
+// Update the Question interface to include the link
+interface Question {
```

Comment:
Renaming to BackendQuestion does not make sense because both the frontend and backend are using TypeScript, allowing the entity to be shared across both layers. Additionally, there should be a DTO (Data Transfer Object) responsible for handling the transmission of data between the frontend and backend to ensure type safety and consistency.